assets/javascripts/discourse/controllers/user-notes.js

```diff
@@ -1,70 +1,80 @@
-import I18n from "I18n";
-import discourseComputed, { on } from "discourse-common/utils/decorators";
-import { popupAjaxError } from "discourse/lib/ajax-error";
 import Controller from "@ember/controller";
+import I18n from "I18n";
+import { action } from "@ember/object";
 import { inject as service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { tracked } from "@glimmer/tracking";
 
-export default Controller.extend({
-  dialog: service(),
-  newNote: null,
-  saving: false,
-  user: null,
+export default class UserNotesController extends Controller {
+  @service dialog;
 
-  @on("init")
-  reset() {
-    this.setProperties({ newNote: null, saving: false, callback: null });
-  },
+  @tracked newNote;
+  @tracked saving;
+  @tracked userId;
+
+  constructor() {
+    super(...arguments);
 
-  @discourseComputed("newNote", "saving")
-  attachDisabled(newNote, saving) {
-    return saving || !newNote || newNote.length === 0;
-  },
+    this.reset();
+  }
```

Comment:
Now that this is a native class, I think we can just initialize the properties when they're initially defined, and get rid of the `reset` method. Looks like all of them get initialized to `null` anyway, so we only need to give `saving` an initial value:

i.e.:

```
@tracked saving = false;
```

Reply:
Ah I see, I didn't realise it was being called from another place. We'll have to keep it 👌

Reply:
Great!